### ISSUE TITLE
Java14 Full Records Support Check Validation: GenericWhitespaceCheck

### ISSUE THREAD

**Issue by nmancus1** [checkstyle/checkstyle#8524]
Child of #8452 
Check documentation: https://checkstyle.sourceforge.io/config_whitespace.html#GenericWhitespace

From check documentation:
_Checks that the whitespace around the Generic tokens (angle brackets) "<" and ">" are correct to the typical convention. The convention is not configurable._

```
➜  full-record-grammar /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 TestClass.java
Note: TestClass.java uses preview language features.
Note: Recompile with -Xlint:preview for details.
➜  full-record-grammar cat config.xml    
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
      <module name="GenericWhitespace"/>
  </module>
</module>
➜  full-record-grammar cat TestClass.java
import java.util.List;

// in record def
record MyTestRecord() {
    static List< String> l; // violation, "<" followed by whitespace
}

//in ctor
record MyTestRecord2() {
    MyTestRecord2(String one, String two, String three){
        this();
        List< String> l; // violation, "<" followed by whitespace
    }
}

//in compact ctor
record MyTestRecord3() {
    public MyTestRecord3{
        List< String> l; // violation, "<" followed by whitespace
    }

}

// in record component
record MyTestRecord4(List< String> l, String two, String three) {
                      // ^^^ violation, "<" followed by whitespace

}

// in record header
record MyTestRecord6< T>(){} // violation, "<" followed by whitespace
record MyTestRecord7<T >(){} // violation, "<" preceded by whitespace

//in ctor params
record MyTestRecord8() {
    MyTestRecord8(List< String> l, String two, String three){
        this();    // ^^^ violation, "<" followed by whitespace
    }
}
➜  full-record-grammar java $RUN_LOCALE -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.35-SNAPSHOT-all.jar -c config.xml TestClass.java
Starting audit...
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:5:16: '<' is followed by whitespace. [GenericWhitespace]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:12:13: '<' is followed by whitespace. [GenericWhitespace]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:19:13: '<' is followed by whitespace. [GenericWhitespace]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:25:26: '<' is followed by whitespace. [GenericWhitespace]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:31:21: '<' is followed by whitespace. [GenericWhitespace]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:32:24: '>' is preceded with whitespace. [GenericWhitespace]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:36:23: '<' is followed by whitespace. [GenericWhitespace]
Audit done.
Checkstyle ends with 7 errors.
```
This check works as intended, even in record components, so I think we can close this issue.

**Comment by pbludov** [comment]
The tricky part here is a generic ctor, like this:
```java
class TestClass {
    <T> TestClass(List<T> arg) { }
}
```
But canonical constructor has no args, so may not have a type variable.
